File: library/src/main/java/zjm/cst/dhu/library/api/repository/BaseRepository.java
package zjm.cst.dhu.library.api.repository;

import java.util.List;
import java.util.Map;

import okhttp3.ResponseBody;
import rx.Observable;
import zjm.cst.dhu.library.mvp.model.NewsData;
import zjm.cst.dhu.library.mvp.model.NewsInf;

/**
 * Created by zjm on 2017/5/10.
 */

public interface BaseRepository {

    Observable<Map<String, List<NewsData>>> getNewsList(String type, String id, int start);

    Observable<Map<String, NewsInf>> getNewInf(String postId);

    Observable<ResponseBody> getNewsInfImg(String path);
}

File: library/src/main/kotlin/zjm/cst/dhu/library/api/repository/BaseRepository.kt
package zjm.cst.dhu.library.api.repository

import okhttp3.ResponseBody
import zjm.cst.dhu.library.mvp.model.NewsData
import zjm.cst.dhu.library.mvp.model.NewsInf
import rx.Observable

/**
 * Created by zjm on 2017/5/10.
 */

interface BaseRepository {

    fun getNewsList(type: String, id: String, start: Int):
            Observable<Map<String, List<NewsData>>>

    fun getNewInf(postId: String):
            Observable<Map<String, NewsInf>>

    fun getNewsInfImg(path: String):
            Observable<ResponseBody>
}